### PR TITLE
Identity v3: 401 is not OK for tokens

### PR DESCRIPTION
**Pull request by jtopjian**
For #388

### CODE REVIEW

**Comment by jtopjian** [comment]
Side thought: If an OpenStack environment was, possibly incorrectly, configured to return a 401 on something other than a token validation (for example, to deny access to a certain feature (such as preventing the creation of Load Balancers)), would that trigger a loop [here](https://github.com/gophercloud/gophercloud/blob/master/provider_client.go#L210)?

File: openstack/identity/v3/tokens/requests.go [hunk 1]
```diff
@@ -182,7 +182,7 @@ func Get(c *gophercloud.ServiceClient, token string) (r GetResult) {
 func Validate(c *gophercloud.ServiceClient, token string) (bool, error) {
 	resp, err := c.Request("HEAD", tokenURL(c), &gophercloud.RequestOpts{
 		MoreHeaders: subjectTokenHeaders(c, token),
-		OkCodes:     []int{200, 204, 400, 401, 403, 404},
+		OkCodes:     []int{200, 204, 400, 403, 404},
```
Comment (jrperritt) [inline openstack/identity/v3/tokens/requests.go]: Yeah, I'm not sure why all these are in here. I think 200, 204, and 404 should be the only ones.

Comment (jtopjian) [inline openstack/identity/v3/tokens/requests.go]: The PR that implemented this is #203. This is was my misunderstanding of `OkCodes`. I was under the impression that any valid response (no matter success or failure), should be listed as a valid code.

**Comment by jrperritt** [comment]
> would that trigger a loop here?

Yes, that's why users should limit re-auth attempts in a custom `http.Client`. See #230.

File: openstack/identity/v3/tokens/requests.go (hunk 1 above)
Comment (jtopjian) [inline openstack/identity/v3/tokens/requests.go]: While we're on the topic (but I can open a separate Issue/PR), the only other place I see 400 codes listed as "OK" are here:

https://github.com/gophercloud/gophercloud/blob/c5c590a7ee87267c978586fdd1655db0dd4b6e05/openstack/imageservice/v2/members/requests.go#L21

Should those be removed, too?

To confirm, an "OK Code" is one that would only signify a successful action? Not one that is necessarily used in the API service itself?

Comment (jrperritt) [inline openstack/identity/v3/tokens/requests.go]: > Should those be removed, too?

Yes

> To confirm, an "OK Code" is one that would only signify a successful action? Not one that is necessarily used in the API service itself?

Correct. In fact, the only reason the `Validate` function accepts 404 is for a legacy reason. It returns a `bool`, and 404 there would signify `false` (i.e. unauthorized).

Comment (jtopjian) [inline openstack/identity/v3/tokens/requests.go]: Perfect - thanks! And the 404 was the odd one out that I was _hoping_ had a special case :)

**Pull request merged**